### PR TITLE
fix(DatePicker, react-datepicker-compat): Make year picker react to go to today button

### LINKED ISSUE #27932: [Bug]: Go to today doesn't work for year picker
### Library

React v8 (@fluentui/react)

### System Info

```shell
NA
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codepen.io/Danavishnu_kb/pen/vYVaaQa

## Actual Behavior
Goto Today option is not working for year picker.

Go to year picker, navigate to 1970 and then select goto today. The component gets stuck in year picker.

## Expected Behavior
Goto Today should work on year picker.

### Logs

_No response_

### Requested priority

Blocking

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by sopranopillow**
## Previous Behavior

Year picker was not updating itself when the navigated year or selected year changed. This made it so that when clicking go to today, the picker would not update to the current year.

https://github.com/microsoft/fluentui/assets/5953191/1bdb48d1-a413-4851-aab2-17d521a3c939

## New Behavior

https://github.com/microsoft/fluentui/assets/5953191/ce87b555-3ff5-4f44-b4c1-7019128d9a13

Year picker's state now takes a look at the navigated and selected date to update the current years in view accordingly.

## Related Issue(s)

- Fixes #27932

### CODE REVIEW

File: packages/react-components/react-datepicker-compat/src/components/CalendarYear/CalendarYear.tsx [hunk 1]
```diff
@@ -377,26 +377,26 @@ function useAnimateBackwards({ selectedYear, navigatedYear }: CalendarYearProps)
...
-  const onNavPrevious = () => navigate(NavigationDirection.Previous);
+  const [fromYear, setFromYear] = React.useState<number>(rangeYear);
+
+  const onNavNext = () => {
+    setFromYear(i => i + CELL_COUNT);
+  };
+
+  const onNavPrevious = () => {
+    setFromYear(i => i - CELL_COUNT);
+  };
```
Comment (khmakoto) [inline packages/react-components/react-datepicker-compat/src/components/CalendarYear/CalendarYear.tsx:393]: _nit:_ Use relevant variable names

```suggestion
  const onNavNext = () => {
    setFromYear(year => year + CELL_COUNT);
  };

  const onNavPrevious = () => {
    setFromYear(year => year - CELL_COUNT);
  };
```

File: packages/react/src/components/Calendar/CalendarYear/CalendarYear.base.tsx [hunk 2]
```diff
@@ -398,26 +398,26 @@ function useAnimateBackwards({ selectedYear, navigatedYear }: ICalendarYearProps
...
-  const onNavPrevious = () => navigate(NavigationDirection.Previous);
+  const [fromYear, setFromYear] = React.useState<number>(rangeYear);
+
+  const onNavNext = () => {
+    setFromYear(i => i + CELL_COUNT);
+  };
+
+  const onNavPrevious = () => {
+    setFromYear(i => i - CELL_COUNT);
+  };
```
Comment (khmakoto) [inline packages/react/src/components/Calendar/CalendarYear/CalendarYear.base.tsx:414]: _nit:_ Use relevant variable names

```suggestion
  const onNavNext = () => {
    setFromYear(year => year + CELL_COUNT);
  };

  const onNavPrevious = () => {
    setFromYear(year => year - CELL_COUNT);
  };
```

File: change/@fluentui-react-a941c131-ba0e-40bc-a4b1-a87cd8554cff.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-a941c131-ba0e-40bc-a4b1-a87cd8554cff.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=28907&commitId=d88f81b8217b7c8de9725e3afcc5fa23d0a6c1f4&env=prod&clientType=FLUENTUI)


  <details><summary><b>Image</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Image.Fit- cover, image taller.default.chromium.png|7603| Changed|
|Image.Fit- contain, image taller.default.chromium.png|1186| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Review by khmakoto — APPROVED** [review]

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                    |     Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                  Change |
| :------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ----------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr> | `207.127 kB`<br />`57.41 kB` | `207.157 kB`<br />`57.399 kB` | `30 B` <br />`-11 B`  |

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/669c13cc90c399d2e0e727be721f868c6c6e2932'>669c13cc90c399d2e0e727be721f868c6c6e2932</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-a941c131-ba0e-40bc-a4b1-a87cd8554cff.json (hunk 3 above)
Comment (fabricteam) [inline change/@fluentui-react-a941c131-ba0e-40bc-a4b1-a87cd8554cff.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=28907&commitId=0b1a52b7b3b398f22e0f0f408eafbf1a1412e282&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 4 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.customSize+image.normal.chromium.png|84| Changed|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|3357| Changed|
|Avatar Converged.size+image+badge.normal.chromium.png|52| Changed|
|Avatar Converged.badgeMask.normal.chromium.png|1340| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**